<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Dynamic Programming</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="0">
         <layout class="QGridLayout" name="gridLayout">
          <item row="5" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Total weight</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QLabel" name="DPTotalProfitLabel">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="DPDataSetComboBox"/>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Solution</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Data set</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="SolutionItemsList" name="DPSolutionTable"/>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="DPSackSizeSpinBox">
            <property name="maximum">
             <number>999999999</number>
            </property>
           </widget>
          </item>
          <item row="7" column="1">
           <widget class="QLabel" name="DPExecutionTimeLabel">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QLabel" name="DPTotalWeightLabel">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>Total profit</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Data:</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="ItemsList" name="DPDataTable"/>
          </item>
          <item row="7" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Execution time</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Sack size</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QPushButton" name="DPExecuteAlgorithmBtn">
            <property name="text">
             <string>Execute algorithm</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Branch and bound</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="0" column="0">
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="5" column="0">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Total weight</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QLabel" name="BBTotalProfitLabel">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="BBDataSetComboBox"/>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>Solution</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_10">
            <property name="text">
             <string>Data set</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="SolutionItemsList" name="BBSolutionTable"/>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="BBSackSizeSpinBox">
            <property name="maximum">
             <number>999999999</number>
            </property>
           </widget>
          </item>
          <item row="7" column="1">
           <widget class="QLabel" name="BBExecutionTimeLabel">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QLabel" name="BBTotalWeightLabel">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="label_11">
            <property name="text">
             <string>Total profit</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_12">
            <property name="text">
             <string>Data:</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="ItemsList" name="BBDataTable"/>
          </item>
          <item row="7" column="0">
           <widget class="QLabel" name="label_13">
            <property name="text">
             <string>Execution time</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_14">
            <property name="text">
             <string>Sack size</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QPushButton" name="BBExecuteAlgorithmBtn">
            <property name="text">
             <string>Execute algorithm</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>Heuristics</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_7">
        <item row="0" column="0">
         <layout class="QGridLayout" name="gridLayout_6">
          <item row="0" column="0">
           <widget class="QLabel" name="label_17">
            <property name="text">
             <string>Data set</string>
            </property>
           </widget>
          </item>
          <item row="8" column="0">
           <widget class="QLabel" name="label_20">
            <property name="text">
             <string>Execution time</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_19">
            <property name="text">
             <string>Data:</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="SolutionItemsList" name="HSolutionTable"/>
          </item>
          <item row="4" column="1">
           <widget class="ItemsList" name="HDataTable"/>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="label_15">
            <property name="text">
             <string>Total weight</string>
            </property>
           </widget>
          </item>
          <item row="8" column="1">
           <widget class="QLabel" name="HExecutionTimeLabel">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_21">
            <property name="text">
             <string>Sack size</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="HDataSetComboBox"/>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="label_16">
            <property name="text">
             <string>Solution</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QPushButton" name="HExecuteAlgorithmBtn">
            <property name="text">
             <string>Execute algorithm</string>
            </property>
           </widget>
          </item>
          <item row="7" column="1">
           <widget class="QLabel" name="HTotalProfitLabel">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="HSackSizeSpinBox">
            <property name="maximum">
             <number>999999999</number>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QLabel" name="HTotalWeightLabel">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="7" column="0">
           <widget class="QLabel" name="label_18">
            <property name="text">
             <string>Total profit</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QComboBox" name="HMethodCB"/>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_22">
            <property name="text">
             <string>Method</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_4">
       <attribute name="title">
        <string>Genetic algorithm</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_9">
        <item row="0" column="0">
         <layout class="QGridLayout" name="gridLayout_8">
          <item row="8" column="1">
           <widget class="QLabel" name="GATotalProfitLabel">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="label_27">
            <property name="text">
             <string>Data:</string>
            </property>
           </widget>
          </item>
          <item row="9" column="1">
           <widget class="QLabel" name="GAExecutionTimeLabel">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QSpinBox" name="GAGenerationsCount">
            <property name="maximum">
             <number>999999999</number>
            </property>
           </widget>
          </item>
          <item row="7" column="1">
           <widget class="QLabel" name="GATotalWeightLabel">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="GASackSizeSpinBox">
            <property name="maximum">
             <number>999999999</number>
            </property>
           </widget>
          </item>
          <item row="8" column="0">
           <widget class="QLabel" name="label_26">
            <property name="text">
             <string>Total profit</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="ItemsList" name="GADataTable"/>
          </item>
          <item row="7" column="0">
           <widget class="QLabel" name="label_23">
            <property name="text">
             <string>Total weight</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QPushButton" name="GAExecuteAlgorithmBtn">
            <property name="text">
             <string>Execute algorithm</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="GADataSetComboBox"/>
          </item>
          <item row="9" column="0">
           <widget class="QLabel" name="label_28">
            <property name="text">
             <string>Execution time</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_31">
            <property name="text">
             <string>Generations count</string>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="label_24">
            <property name="text">
             <string>Solution</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_25">
            <property name="text">
             <string>Data set</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="SolutionItemsList" name="GASolutionTable"/>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="GAPopulationCount">
            <property name="maximum">
             <number>999999999</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_29">
            <property name="text">
             <string>Sack size</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_30">
            <property name="text">
             <string>Population count</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_5">
       <attribute name="title">
        <string>Simulated annealing</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_13">
        <item row="0" column="0">
         <layout class="QGridLayout" name="gridLayout_12">
          <item row="1" column="0">
           <widget class="QLabel" name="label_48">
            <property name="text">
             <string>Sack size</string>
            </property>
           </widget>
          </item>
          <item row="7" column="0">
           <widget class="QLabel" name="label_42">
            <property name="text">
             <string>Total profit</string>
            </property>
           </widget>
          </item>
          <item row="8" column="0">
           <widget class="QLabel" name="label_44">
            <property name="text">
             <string>Execution time</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_41">
            <property name="text">
             <string>Data:</string>
            </property>
           </widget>
          </item>
          <item row="7" column="1">
           <widget class="QLabel" name="SATotalProfitLabel">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_47">
            <property name="text">
             <string>Data set</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="SADataSetComboBox"/>
          </item>
          <item row="5" column="1">
           <widget class="SolutionItemsList" name="SASolutionTable"/>
          </item>
          <item row="6" column="1">
           <widget class="QLabel" name="SATotalWeightLabel">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="SASackSizeSpinBox">
            <property name="maximum">
             <number>999999999</number>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="ItemsList" name="SADataTable"/>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="label_43">
            <property name="text">
             <string>Total weight</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="label_46">
            <property name="text">
             <string>Solution</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QPushButton" name="SAExecuteAlgorithmBtn">
            <property name="text">
             <string>Execute algorithm</string>
            </property>
           </widget>
          </item>
          <item row="8" column="1">
           <widget class="QLabel" name="SAExecutionTimeLabel">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_45">
            <property name="text">
             <string>Iterations count</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="SAIterationsCountSpinBox">
            <property name="maximum">
             <number>999999999</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_6">
       <attribute name="title">
        <string>Hybrid algorithm</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_11">
        <item row="0" column="0">
         <layout class="QGridLayout" name="gridLayout_10">
          <item row="6" column="1">
           <widget class="ItemsList" name="HADataTable"/>
          </item>
          <item row="8" column="0">
           <widget class="QLabel" name="label_34">
            <property name="text">
             <string>Total weight</string>
            </property>
           </widget>
          </item>
          <item row="9" column="0">
           <widget class="QLabel" name="label_33">
            <property name="text">
             <string>Total profit</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_40">
            <property name="text">
             <string>Population count</string>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="label_32">
            <property name="text">
             <string>Data:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_39">
            <property name="text">
             <string>Sack size</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="HASackSizeSpinBox">
            <property name="maximum">
             <number>999999999</number>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="HADataSetComboBox"/>
          </item>
          <item row="3" column="1">
           <widget class="QSpinBox" name="HAGenerationsCount">
            <property name="maximum">
             <number>999999999</number>
            </property>
           </widget>
          </item>
          <item row="8" column="1">
           <widget class="QLabel" name="HATotalWeightLabel">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="10" column="0">
           <widget class="QLabel" name="label_35">
            <property name="text">
             <string>Execution time</string>
            </property>
           </widget>
          </item>
          <item row="10" column="1">
           <widget class="QLabel" name="HAExecutionTimeLabel">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_38">
            <property name="text">
             <string>Data set</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="HAPopulationCount">
            <property name="maximum">
             <number>999999999</number>
            </property>
           </widget>
          </item>
          <item row="7" column="1">
           <widget class="SolutionItemsList" name="HASolutionTable"/>
          </item>
          <item row="7" column="0">
           <widget class="QLabel" name="label_37">
            <property name="text">
             <string>Solution</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QPushButton" name="HAExecuteAlgorithmBtn">
            <property name="text">
             <string>Execute algorithm</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_36">
            <property name="text">
             <string>Generations count</string>
            </property>
           </widget>
          </item>
          <item row="9" column="1">
           <widget class="QLabel" name="HATotalProfitLabel">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QSpinBox" name="HAIterationsCountSpinBox">
            <property name="maximum">
             <number>999999999</number>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_49">
            <property name="text">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Simulated annealing &lt;/p&gt;&lt;p&gt;iterations count&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ItemsList</class>
   <extends>QTableWidget</extends>
   <header>itemslist.h</header>
  </customwidget>
  <customwidget>
   <class>SolutionItemsList</class>
   <extends>QTableWidget</extends>
   <header>solutionitemslist.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
